numpy/core/_add_newdocs.py

```diff
@@ -3938,18 +3938,19 @@
     Construct Python bytes containing the raw data bytes in the array.
 
     Constructs Python bytes showing a copy of the raw contents of
-    data memory. The bytes object can be produced in either 'C' or 'Fortran',
-    or 'Any' order (the default is 'C'-order). 'Any' order means C-order
-    unless the F_CONTIGUOUS flag in the array is set, in which case it
-    means 'Fortran' order.
+    data memory. The bytes object can be produced in C-order by default.
+    This behavior is controlled by the order parameter.
 
     .. versionadded:: 1.9.0
 
     Parameters
     ----------
-    order : {'C', 'F', None}, optional
-        Order of the data for multidimensional arrays:
-        C, Fortran, or the same as for the original array.
+    order : {'C', 'F', 'A', 'K'}, optional
+        Controls the memory layout of the copy. 'C' means C-order,
+        'F' means F-order, 'A' means 'F' if `a` is Fortran contiguous,
+        'C' otherwise. 'K' means match the layout of `a` as closely
```

Comment:
not sure if the following is expected: on a fortran ordered ndarray a: a.tobytes('K') == a.tobytes('F') is False.

Reply:
I think value of a.tobytes('K') would depend on the nature of array, and wouldn't always be False/True when compared with 'F'(column-major). @rossbar Can you please reply here?

Reply:
Good point - there does seem to be a problem here...

Reply:
That seems arguably like a bug in `tobytes`, it simply was written before "K" existed probably.

Reply:
I think we've stumbled onto another issue here - the original docstring only listed 'C', 'Fortran' and 'Any' as options (though `tobytes` does indeed accept `order=K`) which suggests @seberg is correct. I'm not clear on what should be done about it wrt the docs here

Reply:
I had another look... `np.ravel()` does exactly what we need here, so I think this function should be simplified on the C-side to call `PyArray_Ravel()` which also supports `K` correctly, and then we can just keep things exactly as they are. It creates one additional view, but I do not think that matters too much.

Reply:
thanks @seberg for the summary ! do we have a volunteer for the fix ? if not, i am happy to do it.

Reply:
I say go for it @anirudh2290! We'll leave this doc PR until after the changes have been made - please cross-reference this PR from the one that introduces the fix so we can remind ourselves to revisit it.